### PR TITLE
[MISC] Adding documentation about contributing to docs

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

File: python/CONTRIBUTING.md [hunk 1]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
 tox                      # runs 'lint' and 'unit' environments
 ```
 
+## Documentation
+
+Product documentation is currently stored in [Discourse](https://discourse.charmhub.io/t/charmed-spark-k8s-documentation/8963), which represents the current source of truth, as the content there is rendered in Charmhub and the Ubuntu website.
```
Comment (rgildein) [inline python/CONTRIBUTING.md:61]: not-blocking: Documentation stored sound strange to me, maybe we can use hosted or posted.
```suggestion
Product documentation is currently hosted in [Discourse](https://discourse.charmhub.io/t/charmed-spark-k8s-documentation/8963), which represents the current source of truth, as the content there is rendered in Charmhub and the Ubuntu website. 
```

**Review by rgildein — COMMENTED** [review]
Some questions:
1. In the future, how I will not if source of documentation changes from `releases/3.4/docs` to another version? e.g. `release/4.0/docs`.
2. What if I need to add new topics? How it will work without changing anything in docs/index.md?
3. It's nice to know that Gatekeeper can by run by those 3 options, but how it's triggered in this repo? This is not clear to me.

File: python/CONTRIBUTING.md (hunk 1 above)
Comment (izmalk) [reply python/CONTRIBUTING.md:61]: I feel like `stored` is more accurate here. Storing documentation as in the source of truth, not just publishing.

Suggestion:
```suggestion
Product documentation is stored in [Discourse](https://discourse.charmhub.io/t/charmed-spark-k8s-documentation/8963) and published on Charmhub and the Canonical website via Discourse API. 
```

**Comment by deusebio** [comment]
> 1. In the future, how I will not if source of documentation changes from releases/3.4/docs to another version? e.g. release/4.0/docs.

Ideally every version should have their own documentation

> 2. What if I need to add new topics? How it will work without changing anything in docs/index.md?

Yes. As long as you add the new topic in the "main discourse index page", `docs/index.md` will also be updated and the new files will be created. I will update the documentation to make this point more clear. 

> 3. It's nice to know that Gatekeeper can by run by those 3 options, but how it's triggered in this repo? This is not clear to me.

I will provide a link in the text to the CI where this is clear

File: python/CONTRIBUTING.md [hunk 2]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
 tox                      # runs 'lint' and 'unit' environments
 ```
 
+## Documentation
+
+Product documentation is stored in [Discourse](https://discourse.charmhub.io/t/charmed-spark-k8s-documentation/8963) and published on Charmhub and the Canonical website via Discourse API. 
+The documentation is stored in this repository under the release folder (e.g. `releases/3.4/docs`) and it is continuously mirrored by [Discourse Gatekeeper](https://github.com/canonical/discourse-gatekeeper), that takes care of automatically raise a PR whenever changes to Discourse are made.
```
Comment (izmalk) [inline python/CONTRIBUTING.md:62]: We just said it is stored elsewhere. Let's conflate the mirroring and storing here to not use stored again:

```suggestion
The documentation in this repository under the release folder (e.g. `releases/3.4/docs`) is a mirror synched by [Discourse Gatekeeper](https://github.com/canonical/discourse-gatekeeper), that takes care of automatically raising and updating a PR whenever changes to the content on Discourse are made.
```

File: python/CONTRIBUTING.md [hunk 3]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
...
+Although Discourse content can be edited directly, unless the modifications are trivial and obvious (typos, spellings, formatting) we generally recommend to follow a review process:
+
+1. Create a branch (either in the main repo or from a fork) from the current `main` and add/delete/modify documentation files under the `docs` folder.
```
Comment (izmalk) [inline python/CONTRIBUTING.md:65]:
```suggestion
1. Create a branch (either in the main repo or in a fork) from the current `main` and modify documentation files as necessary.
```

File: python/CONTRIBUTING.md [hunk 4]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
...
+2. Raise a PR against the `main` to start the review process, and address the code review within the PR
```
Comment (izmalk) [inline python/CONTRIBUTING.md:66]:
```suggestion
2. Raise a PR against the `main` to start the review process, and conduct the code review within the PR.
```

File: python/CONTRIBUTING.md [hunk 5]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
...
+3. Once the PR is approved and all comments are addressed, the PR should NOT be merged directly, but the modification should be applied to Discourse. If needed, new Discourse topics can be created, and referenced in the navigation table of the main index file on Discourse.
```
Comment (izmalk) [inline python/CONTRIBUTING.md:67]:
```suggestion
3. Once the PR is approved and all comments are addressed, the PR should NOT be merged directly! All the modifications should be applied to Discourse manually. If needed, new Discourse topics can be created, and referenced in the navigation table of the main index file on Discourse.
```

File: python/CONTRIBUTING.md [hunk 6]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
...
+4. Discourse Gatekeeper will take care of raising a new PR or adding new commits to Discourse PRs already opened, tracking the `discourse-gatekeeper/migrate` branch. The [sync_docs.yaml](https://github.com/canonical/spark-k8s-bundle/blob/main/.github/workflows/sync_docs.yaml) GitHub Actions provides further details on the Gatekeeper integration that can be run (a) in a scheduled fashion every day; (b) as part of pull request CI and (c) can be triggered manually. If new topics are referenced in main index file on Discourse, these will be added to `docs/index.md` and the new topics pulled from Discourse.
```
Comment (izmalk) [inline python/CONTRIBUTING.md:68]:
```suggestion
4. Discourse Gatekeeper will raise a new PR or add new commits to an open Discourse PR, tracking the `discourse-gatekeeper/migrate` branch. The [sync_docs.yaml](https://github.com/canonical/spark-k8s-bundle/blob/main/.github/workflows/sync_docs.yaml) GitHub Actions provides further details on the Gatekeeper integration that can be run (a) in a scheduled fashion every night; (b) as a part of pull request CI, and (c) can be triggered manually. If new topics are referenced in the main index file on Discourse, these will be added to `docs/index.md` and the new topics pulled from Discourse.
```

File: python/CONTRIBUTING.md [hunk 7]
```diff
@@ -56,6 +56,19 @@ tox run -e all-tests     # unit+integration tests
...
+5. Once Gatekeeper has raised or updated an existing PR, feel free to close the initial PR created in step 2, with a comment referring the PR controlled by Gatekeeper. If the initial PR was referring a ticket, add the ticket to either the title or the description of the GateKeeper PR.
```
Comment (izmalk) [inline python/CONTRIBUTING.md:69]:
```suggestion
5. Once Gatekeeper has raised a new or updated an existing PR, feel free to close the initial PR manually created in step 2, with a comment referring to the PR created by Gatekeeper. If the initial PR was referring to a ticket, add the ticket to either the title or the description of the GateKeeper PR.
```

**Review by izmalk — APPROVED** [review]
Looks good. I've added a few suggestions for cosmetic improvements.

**Pull request merged**